### DISCUSSION TITLE
Redirection back to source url after completing checkout?

### DISCUSSION THREAD

**Discussion by baranelitez** [Shopify/storefront-api-feedback#252] [Help] [answered]
Hi. The checkoutUrl given by the graphql API after a cart is filled and ready to be checked out works fine. but the "continue shopping" button at the end of the user flow after payment takes them to the "online store" storefront instead of the origin URL.

Is there a way to fix it? or does anyone know of a workaround?

**Comment by corygibbons** [comment, marked as answer]
I typically setup a redirect in the Online Store theme. Shopify has an example theme here: https://github.com/shopify/hydrogen-redirect-theme
